### PR TITLE
JsonConverter to improve handling of System.Reflection objects

### PR DESCRIPTION
**Pull request by snakefoot**
Protect against System.Reflection Objects like `Assembly`, `MethodInfo`, `Module`, that will dump all types in the the application-assembly.

### CODE REVIEW

File: src/Core/Shipping/BulkSender.cs [hunk 1]
```diff
@@ -36,11 +43,11 @@ public Task<HttpResponseMessage> SendAsync(ICollection<LogzioLoggingEvent> logz,
                 return Task.FromResult(new HttpResponseMessage(System.Net.HttpStatusCode.OK));
 
             var url = string.Format(UrlTemplate, options.ListenerUrl, options.Token, options.Type);
-            var body = SerializeLogEvents(logz, _encodingUtf8, options.UseGzip);
+            var body = SerializeLogEvents(logz, _encodingUtf8);
             return _httpClient.PostAsync(url, body, _encodingUtf8, options.UseGzip);
         }
 
-        private MemoryStream SerializeLogEvents(ICollection<LogzioLoggingEvent> logz, System.Text.Encoding encodingUtf8, bool useGzip)
+        private MemoryStream SerializeLogEvents(ICollection<LogzioLoggingEvent> logz, System.Text.Encoding encodingUtf8)
```
Comment (idohalevi) [inline src/Core/Shipping/BulkSender.cs:50]: @snakefoot Why did you drop gzip support?

Comment (snakefoot) [reply src/Core/Shipping/BulkSender.cs:50]: I just removed an unused parameter. The gzip support handling is not done here.

Comment (snakefoot) [reply src/Core/Shipping/BulkSender.cs:50]: Here the gzip-support is handled:

https://github.com/logzio/logzio-dotnet/blob/c9b5e6c67548b34ee32d820105550b3971cd7ebb/src/Core/HttpClient/IHttpClient.cs#L47-L57

**Comment by idohalevi** [comment]
@yyyogev can you review it?

**Comment by snakefoot** [comment]
@idohalevi Guess one review is enough then :). Time to merge?

**Comment by yyyogev** [comment]
@snakefoot sorry for the delay, we had many holidays these last 2 weeks. Will review it soon

**Pull request merged**